Add support to cuddle assignments for a whole block (#163)
bombsimon/wsl: doc/configuration.md

)
```

> ### allow-cuddle-used-in-block

Author: I was unsure of what to link this header to since in the `rules.md` file there isn't any specific mention of "you can only cuddle blocks with vars used in the first line". lmk if you have an opinions

Reviewer: This is fine, no strong opinion on where it should go. Yeah it's hard to link to `rules.md` since it affects so many. The rules about cuddling used in block should've been a single rule and applied to all blocks, currently the diagnostics are _very_ different and doesn't follow any pattern which is a shame. They will be rewritten in a future major release.

> Controls if you may cuddle variables used in block statements. This is false by default as to not change existing behavior.

Reviewer: After this is merged the part about not changing behavior will no longer make much sense so we can just drop it. Also the fault is shown right under. Maybe clarify to say it's anywhere within the immediate following block?

```suggestion
Controls if you may cuddle variables used anywhere in the following block.
```
